/*
 * This file was automatically generated by EvoSuite
 * Sun Nov 21 02:16:54 GMT 2021
 */

package com.facebook.presto.cassandra;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.facebook.presto.cassandra.CassandraConnectorFactory;
import com.facebook.presto.spi.ConnectorHandleResolver;
import com.facebook.presto.spi.connector.ConnectorContext;
import java.util.Map;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class CassandraConnectorFactory_ESTest extends CassandraConnectorFactory_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      CassandraConnectorFactory cassandraConnectorFactory0 = null;
      try {
        cassandraConnectorFactory0 = new CassandraConnectorFactory("");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // name is null or empty
         //
         verifyException("com.google.common.base.Preconditions", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      CassandraConnectorFactory cassandraConnectorFactory0 = new CassandraConnectorFactory("confiH  ~Vl");
      ConnectorHandleResolver connectorHandleResolver0 = cassandraConnectorFactory0.getHandleResolver();
      assertNotNull(connectorHandleResolver0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      CassandraConnectorFactory cassandraConnectorFactory0 = new CassandraConnectorFactory("confiH  ~Vl");
      ConnectorContext connectorContext0 = mock(ConnectorContext.class, new ViolatedAssumptionAnswer());
      // Undeclared exception!
      try { 
        cassandraConnectorFactory0.create("confiH  ~Vl", (Map<String, String>) null, connectorContext0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // config is null
         //
         verifyException("java.util.Objects", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      CassandraConnectorFactory cassandraConnectorFactory0 = new CassandraConnectorFactory("confiH  ~Vl");
      String string0 = cassandraConnectorFactory0.getName();
      assertEquals("confiH  ~Vl", string0);
  }
}
